resources/components/ImageCard.vue

```diff
@@ -97,6 +106,19 @@ module.exports = {
 			return this.image.descriptionurl;
 		},
 
+		publishDisabled: function () {
+			return this.confirmedSuggestions.length < 1;
+		},
+
+		suggestions: function () {
```

Comment:
The ImageCard component receives suggestions as part of the Image prop that gets passed down to it from the parent. Then it takes some data out of this prop (suggestions) and mutates it as the user toggles the confirmed state on individual Suggestion items.

The code here works, but it has some side-effects that may lead to unintended behavior down the road. Normally a component is not allowed to mutate something it receives as a prop from its parent. But if the prop is an object as opposed to a simple value like a string or number, you actually **can** mutate the values of that object's properties. However, doing so also mutates the state of the parent where this prop originated. This is old-school "two-way data binding", and it's a violation of expected patterns of parent-child communication. It doesn't directly cause any problems here, but if there was a sibling component that also displayed suggestion data somehow, we might see some unexpected behavior that would be hard to track down.

I was trying to get around this previously by moving `suggestions` into state, but I naively assumed that I could write `suggestions: this.image.suggestions` inside of `data()` and get a new copy of the value, but really I got a reference. So the previous version of the code here was also unintentionally modifying parent state when suggestions were toggled.

There are two "best-practice" solutions for cases when the parent passes data down to a child that is then expected to change it somehow:

1. The child doesn't actually change anything, it just emits events that tell the parent what needs to be changed
2. The child creates a true local copy of the data we care about in its state using some kind of object clone method. This is a little more involved because we need to also `watch` for changes in the props that are passed down so we can overwrite the previous cloned data with new cloned data. The initial cloning could happen in a lifecycle hook (maybe `created()`).

Since we want this app to be a showcase for how to use Vue, I think we should change how we're handling data here to one of the above approaches so that we don't unintentionally demonstrate anti-patterns. It seems like the second approach may be better here, since I figure we want to keep most of the logic in ImageCard itself.

Reply:
Just to be clear, we need to make a true copy of `image` before filtering `image.suggestions`? I thought filtering that array would be enough to create a new variable instead of referencing the original image object.

Reply:
We need to copy all of the suggestions before we start mutating them, since we're changing the data of an individual suggestion directly when it's clicked. I don't think we need to copy other properties of `image` here, just the things we are going to change.